scripts/environment_json.sh

```diff
@@ -41,74 +47,26 @@ while getopts rp:h? opt; do
 done
 shift $((OPTIND-1))
 
-function pull_from_config_xml() {
-  cat << EOM | node
-const fs = require('fs');
-const xml2js = require('xml2js');
-const s = fs.readFileSync('config.xml');
-xml2js.parseString(s, function(err, result) {
-  console.log($1);
-});
-EOM
-}
+if [[ "${TYPE}" == "release" && -z ${SENTRY_DSN:-} ]]; then
+  echo "SENTRY_DSN is undefined." 1>&2
+  exit 1
+fi
 
-function pull_from_plist() {
-  cat << EOM | node
-const fs = require('fs');
-const xml2js = require('xml2js');
-const s = fs.readFileSync('$1');
-xml2js.parseString(s, function(err, result) {
-  const dict = result['plist']['dict'][0];
-  const keys = dict['key'];
-  const values = dict['string'];
-  console.log(values[keys.indexOf('$2')]);
-});
-EOM
-}
+APP_VERSION="$(node "$(dirname "$0")/get_version.mjs" "${PLATFORM}")"
+APP_BUILD_NUMBER="$(node "$(dirname "$0")/get_build_number.mjs" "${PLATFORM}")"
 
-function pull_from_ios_plist() {
-  pull_from_plist "apple/xcode/ios/Outline/Outline-Info.plist" $1
-}
+environment=()
 
-function pull_from_osx_plist() {
-  pull_from_plist "apple/xcode/osx/Outline/Outline-Info.plist" $1
-}
-
-function validate_env_vars() {
-  if [[ -z ${SENTRY_DSN:-} ]]; then
-    echo "SENTRY_DSN is undefined." 1>&2
-    exit 1
-  fi
-}
+if [[ ! -z ${SENTRY_DSN:-} ]]; then
```

Comment:
if there's a cleaner way to do this, please help me out 😅

Reply:
I think this is fine, though writing this file in Node.js would probably be cleaner.

I think I'd merge this with the previous SENTRY_DSN block so it's tested only once, like:
```
if SENTRY_DNS:
  add to config
else 
  if release: error
end
```